GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -1459,6 +1463,28 @@ private void ResetSelectedItemsWithConfirmation(bool resetToParent)
 
         private static bool RenamedIndexItem(FileStatusItem item) => item.Item.IsRenamed && item.Item.Staged == StagedStatus.Index;
 
+        List<string> IRunScript.IFileListSource.GetFiles()
+        {
+            IReadOnlyList<FileStatusItem> selectedItems = DiffFiles.SelectedItems.ToList();
+
+            if (!selectedItems.Any())
+            {
+                return null;
+            }
+
+            return selectedItems.Select(item => _fullPathResolver.Resolve(item.Item.Name)).ToList();
+        }
```

Comment:
We're doing multiple allocations here, which are unnecessary.
```suggestion
        IEnumerable<string> IRunScript.IFileListSource.GetFiles() => selectedItems.Select(item => _fullPathResolver.Resolve(item.Item.Name));
```

Reply:
Accepted